management/http.go

```diff
@@ -109,16 +117,66 @@ func (client *Client) createHttpClient() *http.Client {
 	ssl := &tls.Config{}
 	ssl.Certificates = []tls.Certificate{cert}
 
+	var requestLogger *log.Logger
+	if os.Getenv(EnvEnableRequestLogs) != "" {
+		out := os.Stderr
+		if logfile := os.Getenv(EnvRequestLogPath); logfile != "" {
+			var err error
+			out, err = os.OpenFile(logfile, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
+			if err != nil {
+				panic(err)
+			}
+		}
+		requestLogger = log.New(out, "azuresdk.management", log.LstdFlags)
+	}
+
 	httpClient := &http.Client{
-		Transport: &http.Transport{
+		Transport: &loggedTransport{logger: requestLogger, inner: &http.Transport{
 			Proxy:           http.ProxyFromEnvironment,
 			TLSClientConfig: ssl,
-		},
+		}},
 	}
 
 	return httpClient
 }
 
+type loggedTransport struct {
+	logger *log.Logger
+	inner  *http.Transport
+}
+
+func (l *loggedTransport) RoundTrip(request *http.Request) (response *http.Response, err error) {
+	// log request
+	if l.logger != nil {
```

Comment:
bad practice. have a nil logger instead of nil checks. http://en.wikipedia.org/wiki/Null_Object_pattern